Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Collections/GridObjectCollection.cs

```diff
@@ -0,0 +1,419 @@
+                            ((i * CellWidth) - offsetX + halfCell.x) + NodeList[cellCounter].Offset.x,
+                            (-(j * CellHeight) + offsetY - halfCell.y) + NodeList[cellCounter].Offset.y,
+                            0.0f
+                            );
+                    }
+                    cellCounter++;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Update the facing of a node given the nodes new position for facing orign with node and orientation type
+        /// </summary>
+        /// <param name="node"></param>
+        /// <param name="orientType"></param>
+        /// <param name="newPos"></param>
+        private void UpdateNodeFacing(ObjectCollectionNode node)
+        {
+            Vector3 centerAxis;
+            Vector3 pointOnAxisNearestNode;
+            switch (OrientType)
+            {
+                case OrientationTypeEnum.FaceOrigin:
+                    node.transform.rotation = Quaternion.LookRotation(node.transform.position - transform.position, transform.up);
+                    break;
+
+                case OrientationTypeEnum.FaceOriginReversed:
+                    node.transform.rotation = Quaternion.LookRotation(transform.position - node.transform.position, transform.up);
+                    break;
+
+                case OrientationTypeEnum.FaceCenterAxis:
+                    centerAxis = Vector3.Project(node.transform.position - transform.position, transform.up);
+                    pointOnAxisNearestNode = transform.position + centerAxis;
+                    node.transform.rotation = Quaternion.LookRotation(node.transform.position - pointOnAxisNearestNode, transform.up);
+                    break;
+
+                case OrientationTypeEnum.FaceCenterAxisReversed:
+                    centerAxis = Vector3.Project(node.transform.position - transform.position, transform.up);
+                    pointOnAxisNearestNode = transform.position + centerAxis;
+                    node.transform.rotation = Quaternion.LookRotation(pointOnAxisNearestNode - node.transform.position, transform.up);
+                    break;
+
+                case OrientationTypeEnum.FaceFoward:
+                    node.transform.forward = transform.rotation * Vector3.forward;
+                    break;
+
+                case OrientationTypeEnum.FaceForwardReversed:
+                    node.transform.forward = transform.rotation * Vector3.back;
+                    break;
+
+                case OrientationTypeEnum.FaceParentUp:
+                    node.transform.forward = transform.rotation * Vector3.up;
+                    break;
+
+                case OrientationTypeEnum.FaceParentDown:
+                    node.transform.forward = transform.rotation * Vector3.down;
+                    break;
+
+                case OrientationTypeEnum.None:
+                    break;
+
+                default:
+                    throw new ArgumentOutOfRangeException();
```

Comment:
Don't use Throw, as it can case Unity to crash which makes it really hard to debug.

Use 
```Debug.LogWarning`` or ```Debug.LogError`` instead (depending on the seriousness of the error.
Include help text in the error where possible to ease diagnosis